src/aggregator/aggregator/aggregator.go

```diff
@@ -274,7 +275,8 @@ func (agg *aggregator) AddTimedWithStagedMetadatas(
 	sw := agg.metrics.addTimed.SuccessLatencyStopwatch()
 	agg.updateStagedMetadatas(metas)
 	agg.metrics.timed.Inc(1)
-	shard, err := agg.shardFor(metric.ID)
+	shardingId, _ := parsers.GetMetricIDWithoutLe(metric.ID)
+	shard, err := agg.shardFor(shardingId)
```

Comment:
Want to suggest a hacky alternative for consideration.
Write our own `shardFor` function. (here `shardFor` point to a function).
```
shardForWithoutLe(id byte[]) {
  // Look at the encoding of `id` and search for `le=` pattern.
  start := string.Index(id, "\0\1le=")
  if start < 0 {
    // No "le="
    return hash(id)
  }
  end := // to the end.
  return hash(id[0:start]) + hash(id[end:])
}
```